internal/docker/client.go

```diff
@@ -48,10 +52,33 @@ func GetClient() (*client.Client, error) {
 		}
 	}
 
+	// This tries to create a docker client with the default options
 	dockerClient, err := client.NewClientWithOpts(clientOpts...)
-	if err != nil {
-		return nil, fmt.Errorf("failed create docker client: %w", err)
+	if err == nil {
+		_, err = dockerClient.Ping(context.Background())
+		if err == nil {
+			return dockerClient, nil // Successfully connected
+		}
+	}
+
+	// If on macOS and the default Unix socket didn't work, try the macOS-specific path
+	if runtime.GOOS == "darwin" {
+		user, err := user.Current()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get current user: %w", err)
+		}
+
+		macOSSocketPath := filepath.Join(user.HomeDir, "Library/Containers/com.docker.docker/Data/docker.raw.sock")
```

Comment:
`user.Homedir` has some edge cases that have bitten me on previous projects. I believe we used https://github.com/mitchellh/go-homedir for a fix.